abstractions/php/src/RequestInfo.php

```diff
@@ -0,0 +1,19 @@
+<?php
+namespace Microsoft\Kiota\Abstractions;
+
+use Psr\Http\Message\StreamInterface;
+
+abstract class RequestInfo {
```

Comment:
Leverage PSR-7 instead of defining this for ourselves...

https://github.com/guzzle/psr7/blob/master/src/Request.php
https://github.com/php-fig/http-message/blob/master/src/RequestInterface.php

Reply:
this is meant to be abstract from any HTTP library. My understanding is that by using PSR7 we'd tie the generation result to a specific HTTP lib which would go against Kiota's principles.

Reply:
PSR7 does the opposite. It defines a generic HTTP request interface that any HTTP library can/should use.

Reply:
I understand that PSR7 is a spec for HTTP messaging in PhP according to HTTP/URI specifications.
But from what I understand, guzzle is one implementation of it, the core constructs haven't been defined at part of PhP itself.
> This repository contains a full PSR-7 message implementation, several stream decorators, and some helpful functionality like query string parsing.

Also, the generator expects this generic interface, significant rework might be required to "make things fit". And we want consistency across languages.

Reply:
Ok, I now get what you meant by HTTP lib specific implementation.